enamel/cmd/task_processor.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import os
+import sys
+import signal
+
+import daemon
+import extras
+from oslo_config import cfg
+from oslo_log import log as logging
+
+# NOTE(jaypipes): This code taken from Zuul
+# as of python-daemon 1.6 it doesn't bundle pidlockfile anymore
+# instead it depends on lockfile-0.9.1 which uses pidfile.
+pid_file_module = extras.try_imports(['daemon.pidlockfile', 'daemon.pidfile'])
+
+from enamel import opts
+import enamel.task_processor
+
+LOG = logging.getLogger(__name__)
+
+
+class TaskProcessorApp(object):
+
+    def __init__(self, conf=None):
+        self.conf = conf
+
+    def parse_arguments(self):
+        parser = argparse.ArgumentParser(description='Enamel task processor.')
+        parser.add_argument('--no-daemon', action='store_true',
+                            default=False, help='Do not daemonize.')
+        return parser.parse_args()
+
+    def exit_handler(self, signum, frame):
+        signal.signal(signal.SIGUSR1, signal.SIG_IGN)
+        self.server.stop()
+
+    def run(self):
+        self.server = enamel.task_processor.TaskProcessor(self.conf)
+        self.server.start()
+
+        signal.signal(signal.SIGUSR1, self.exit_handler)
```

Comment:
You sure you want usr1 for exit? Seems outside the norms? I would think SIGTERM here, SIGHUP for reload config, USR1 for _something _special_.

Otherwise this stuff looks teh ossum.